src/components/events/EventForm.jsx

```diff
@@ -58,14 +60,15 @@ class EventForm extends React.Component {
     const { eventDate, eventTitle, numJudges } = this.state;
     const collectionName = 'events';
     const data = {
-      eventDate: new Date(moment(eventDate, 'DD/MM/YYYY').format('MM-DD-YYYY')),
+      eventDate: moment(eventDate).format('YYYY-MM-DD HH:mm:ss.SSS'),
       eventTitle,
       numJudges
     };
     if (formType === dialogType.NEW) {
       await addData(collectionName, data);
     } else {
-      await updateData(collectionName, eventId, data);
+	  await updateEvent(eventId, data);
+	  window.location.reload();
```

Comment:
- add callback function to your API call
- resolve with .then
- manage page reloads through setState -> this updates the data and forces a rerender with the new data